### PR TITLE
NEW: Puppet Server and Puppet DB basic monitoring

### PR DESCRIPTION
**Pull request by andvgal**
- Added universal module based on Puppet status API
- There is room for further improvement to increase metric count

### CODE REVIEW

File: python.d/puppet.chart.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            ["jvm_heap_init", 'initial', "absolute", 1, 1048576, 'hidden'],
+        ]
+    },
+    'jvm_nonheap': {
+        'options': [None, "JVM Non-Heap", "MB", "resources", "puppet.jvm", "area"],
+        'lines': [
+            ["jvm_nonheap_max", 'max', "absolute", 1, 1048576, 'hidden'],
+            ["jvm_nonheap_committed", 'committed', "absolute", 1, 1048576],
+            ["jvm_nonheap_used", 'used', "absolute", 1, 1048576],
+            ["jvm_nonheap_init", 'initial', "absolute", 1, 1048576, 'hidden'],
```
Comment (l2isbad) [inline python.d/puppet.chart.py]: About all those `hidden`. You need them for alarms, right? Not a big deal but charts has [variables](https://github.com/firehol/netdata/wiki/External-Plugins#variable) for such cases (netdata doesn't Initializing db file for variable).

```
'jvm_nonheap': {
    'options': [None, "JVM Non-Heap", "MB", "resources", "puppet.jvm", "area"],
     'lines': [
        ["jvm_nonheap_committed", 'committed', "absolute", 1, 1048576],
    ]
    'variables': [
                ["jvm_nonheap_max"],
    ]
```

(yes, variable has no multiplier, divisor)

Comment (andvgal) [reply python.d/puppet.chart.py]: Yeah, I've been considering variables, but understood the code wrong without digging further. Got initial impression that they are set only at chart creation time. Fixed.

**Comment by andvgal** [comment]
Fixed issues. Ready for review.

File: python.d/puppet.chart.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        ],
+    },
+    'jvm_nonheap': {
+        'options': [None, "JVM Non-Heap", "MB", "resources", "puppet.jvm", "area"],
+        'lines': [
+            ["jvm_nonheap_committed", 'committed', "absolute", 1, MB],
+            ["jvm_nonheap_used", 'used', "absolute", 1, MB],
+        ],
+        'variables': [
+            ['jvm_heap_max', 'jvm_heap_init'],
```
Comment (l2isbad) [inline python.d/puppet.chart.py]: should be a list of variables
```
[
  ['jvm_heap_max'],
  [jvm_heap_init']
]
```

File: python.d/puppet.chart.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        'options': [None, "CPU usage", "percentage", "resources", "puppet.cpu", "stacked"],
+        'lines': [
+            ["cpu_time", 'execution', "absolute", 1, CPU_SCALE],
+            ["gc_time", 'GC', "absolute", 1, CPU_SCALE],
+        ]
+    },
+    'fd_open': {
+        'options': [None, "File Descriptors", "descriptors", "resources", "puppet.fdopen", "line"],
+        'lines': [
+            ["fd_max", 'max', "absolute", 1, 1, 'hidden'],
```
Comment (l2isbad) [inline python.d/puppet.chart.py]: another `hidden`

File: python.d/puppet.chart.py (hunk 2 above)
Comment (andvgal) [reply python.d/puppet.chart.py]: Fixed

File: python.d/puppet.chart.py (hunk 3 above)
Comment (andvgal) [reply python.d/puppet.chart.py]: Fixed

**Comment by l2isbad** [comment]
@andvgal :+1: 
after the merge please update this [wik page](https://github.com/firehol/netdata/wiki/Add-more-charts-to-netdata) (open to everyone for edit)

**Comment by andvgal** [comment]
@l2isbad 
Ehm, it appears I updated Wiki before merge is complete. However, the same situation is with current java.d plugins.

**Comment by l2isbad** [comment]
Oh, i forgot about sidebar (add to **Python Modules** list) :smile:

**Comment by andvgal** [comment]
Done.
Guys, you need some sort of checklist ;))

**Comment by ktsaou** [comment]
merged it! Thank you @andvgal !

**Pull request merged**